generate_release_notes.py

```diff
@@ -125,180 +149,258 @@ def get_repo_list(github_org, github):
+    """
     try:
-        release_notes = "# Releases"
+        github = Github(parsed_args.github_token)
+    except GithubException as error:
+        logging.error("%s", error)
+        sys.exit(1)
 
-        for org in args.organizations:
+    try:
+        notes = "# Releases"
+        for org in parsed_args.organizations:
             for github_repo in get_repo_list(org, github):
-                logging.info("Processing " + github_repo)
+                logging.info("Processing %s", github_repo)
                 repo = get_repo(github_repo, github)
-                # Skip archived and blacklisted repos
+
                 if repo.archived or is_blacklisted_repo(github_repo):
                     continue
-                # Compare branches (release or any branches vs. stable )
-                compare_branches = [args.release_branch, args.stable_branch]
+
+                compare_branches = [parsed_args.release_branch, parsed_args.stable_branch]
                 skip = False
                 for branch in compare_branches:
                     if not is_branch(branch, repo):
-                        logging.warning(github_repo + f": \"{branch}\" branch does not exist, skipping repo")
+                        logging.warning('%s: "%s" branch does not exist, skipping repo', github_repo, branch)
                         skipped_repos.append(f"{github_repo}: missing branch {branch}")
                         skip = True
                         break
 
-                # Skip repo if branch does not exist
                 if skip:
                     continue
 
-                # Get commits
                 found_commits = repo.compare(compare_branches[1], compare_branches[0]).commits
                 commit_only = set()
                 prr_list = set()
-                # Fetch Pull Requests from commits
+                pr_mapping = {}
+
                 for found_commit in found_commits:
                     try:
                         pr_list = found_commit.get_pulls()
                         if pr_list:
                             prr_list.update(list(pr_list))
-                    except Exception as e:
-                        logging.warning(f"Failed to retrieve pull requests for commit {found_commit.sha}: {e}")
+                    except GithubException as error:
+                        logging.warning(
+                            "Failed to retrieve pull requests for commit %s: %s",
+                            found_commit.sha, error)
                         skipped_prs.append(f"Commit {found_commit.sha} in repo {github_repo} could not retrieve PRs")
                     else:
-                        commit_url = found_commit.commit.html_url[:-34]
-                        commit_title = "{} (Commit [{}])".format(found_commit.commit.message.strip().split('\n', 1)[0], found_commit.commit.sha[:6])
+                        commit_title = "{} (Commit [{}])".format(
```

Comment:
Why do we need the individual commit titles. I assumed we would only need Jira or PR titles.

Reply:
We are included commit titles in the orphaned commit detials section to capture changes made to the branch that aren't associated to a PR or Jira issue.